tests/dashboard/test_permissions.py

```diff
@@ -693,22 +707,38 @@ def test_group_detail_form_valid(
     assert staff_group.permissions.get(pk=permission_view_product.pk)
 
 
-def test_user_group_form_not_valid(db):
+def test_staff_form_not_valid(db):
     data = {'groups': 1}
-    form = UserGroupForm(data=data)
+    form = StaffForm(data=data)
     assert not form.is_valid()
 
 
-def test_user_group_form_create_valid(
+def test_staff_form_create_valid(
         admin_client, staff_user, staff_group):
     url = reverse('dashboard:staff-details', args=[staff_user.pk])
-    data = {'groups': staff_group.pk}
+    data = {'email': 'staff@example.com', 'groups': staff_group.pk}
     admin_client.post(url, data)
     staff_user = User.objects.get(pk=staff_user.pk)
     assert staff_user.groups.count() == 1
 
 
-def test_user_group_form_create_not_valid(admin_client, staff_user):
+def test_staff_create_email_with_set_link_password(
+        admin_client, staff_group):
+    url = reverse('dashboard:staff-create')
+    data = {'email': 'staff3@example.com', 'groups': staff_group.pk,
+            'is_staff': True}
+    response = admin_client.post(url, data)
+    assert User.objects.count() == 2
+    assert len(mail.outbox) == 1
+    assert response['Location'] == reverse('dashboard:staff-list')
+
+    match = re.search('http.*', mail.outbox[0].body)
```

Comment:
Instead of matching link in email body, I would rather expect a separate test that constructs this link and checks if the view does what it is expected.

Reply:
The point of this test was to , create new staff member, generate email, go to the location from link and try to set password.
It is still in progress.